tests/test_imageDecorrelation.py

```diff
@@ -347,6 +399,38 @@ def testDiffimCorrection(self):
         # Template variance is higher than that of the science img.
         self._testDiffimCorrection(svar=0.04, tvar=0.08)
 
+    def testNoiseDiffimCorrection(self):
+        """Test correction by estimating correlation directly on a noise difference image.
+
+        Notes
+        ------
+
+        See `lsst-dm/diffimTests` notebook `DM-24371_correlation_estimate.ipynb`
+        for further details of how the correlation looks like in the uncorrected
+        and corrected cases and where the tolerance numbers come from.
+        """
+        svar = 1.
+        tvar = 100.
+        self._setUpSourcelessImages(svar=svar, tvar=tvar)
+        diffExp, mKernel, expected_var = self._makeAndTestUncorrectedDiffim()
+        corrected_diffExp = self._runDecorrelationTask(diffExp, mKernel)
+
+        rho_sci = estimatePixelCorrelation(self.im1ex.getImage().getArray())
+        rho_rawdiff = estimatePixelCorrelation(diffExp.getImage().getArray())
+        rho_corrdiff = estimatePixelCorrelation(corrected_diffExp.getImage().getArray())
+
+        self.assertFloatsAlmostEqual(rho_sci[0], 1., atol=0.1, rtol=None)
+        self.assertFloatsAlmostEqual(rho_rawdiff[0], 1., atol=0.1, rtol=None)
+        self.assertFloatsAlmostEqual(rho_corrdiff[0], 1., atol=0.1, rtol=None)
+
+        self.assertFloatsAlmostEqual(rho_sci[1:], 0., atol=0.1, rtol=None)
+
+        self.assertGreater(rho_rawdiff[1], 0.2)
+        self.assertGreater(rho_rawdiff[2], 0.2)
+        self.assertGreater(rho_rawdiff[3], 0.2)
+
+        self.assertFloatsAlmostEqual(rho_corrdiff[1:], 0., atol=0.1, rtol=None)
```

Comment:
Please define variables with meaningful names to hold the numbers used in the comparisons. Or, add a comment explaining why we should expect, e.g. `rho_rawdiff[1]` to always be greater than 0.2